### ISSUE TITLE
CondEqual should give an error when constructed with vector length > 1

### ISSUE THREAD

**Issue by mb706** [mlr-org/paradox#261]
```r
CondEqual$new(c("b", "c")))
```
should throw an error

**Comment by berndbischl** [comment]
not necessarily, unfortunately. it basically depends on whether we allow the parent param to be ParamUty (hence to use deps in learners). 

*if* we allow that, such a param could have a vector-valued setting. i guess we don't want that, but this then need to make clear asserted

**Comment by mb706** [comment]
In that case the CondEqual test function is broken (should be `identical` instead of `==`)

**Comment by mb706** [comment]
or possibly `isTRUE(all.equal(..))`
